### DISCUSSION TITLE
Does react-spring support  a web animation API (WAAPI) driver?

### DISCUSSION THREAD

**Discussion by DamianPereira** [pmndrs/react-spring#1491] [Support]
Hi! I was reading this article: [Why React needed yet another animation library. Introducing: react-spring](https://blog.usejournal.com/why-react-needed-yet-another-animation-library-introducing-react-spring-8212e424c5ce)

And noticed this part:

> Animated is equipped to handle full-native drivers as well, so in the future, should web animations gain more support, it would even be possible to do all the animation in the compositor thread.

Is this on the roadmap? Or even already implemented?

I'm working on a react app and I'm comparing a native web animation api + css transition approach vs react-spring, I'd love to use the spring motion instead of css easing functions, but Ideally I'd like to not block the js main thread with animations.

**Comment by joshuaellis** [comment]
That article is from 3 years ago.

**Reply by DamianPereira** [reply]
I know, web animation APIs were not widely supported back then, but now they are, so maybe it makes sense to try the approach hinted at in the article in the future? Or did anything change making that quote irrelevant to today?.

**Reply by joshuaellis** [reply]
Oh I see, this isnt on a roadmap right not and no it has not been implemented.

**Reply by joshuaellis** [reply]
But maybe it could be, if you were interested in looking into what it would all mean for the library and create an issue, i'd be happy to read your work and get it on said roadmap

**Reply by DamianPereira** [reply]
I'm not familiar enough with the code/project right now, but if I end up using it instead of the native css + waapi approach, I will look into it (since after using it I'll be a bit more familiar with the whole thing). Thanks!
